### PR TITLE
Get rid of audiokiller.

### PR DESCRIPTION
**Pull request by Lord-Kamina**
### What does this PR do?

Removes the audiokiller hack when closing portaudio streams.

### Closes Issue(s)

#672 

### Motivation

This was a ten-year old hack, and I doubt it's still necessary.
### More

- [ ] Added/updated documentation

### Additional Notes

Please test this!

### CODE REVIEW

**Comment by Baklap4** [comment]
Hmm i tried exactly this tonight already.. didn't work i'm afraid:
```
songs/notice: Done Loading. Loaded 76 Songs.
songs/notice: Done Caching.
songs/notice: Done Caching.
profiler-mainloop/debug:  draw (1.4 ms)  swap (0.3 ms)  textures (0.1 ms)  events (0.0 ms)  fpsctrl (0.0 ms)  misc (0.0 ms)
Saved configuration to "C:\\Users\\Arjan\\AppData\\Roaming\\performous\\config.xml"
pm_winmm_term called
pm_winmm_term exiting
TerminateHostApis in
ERROR: failed PaWinUtil_CoUninitialize calling thread[18408] does not match initializing thread[4972]
ERROR: failed PaWinUtil_CoUninitialize calling thread[18408] does not match initializing thread[4972]
TerminateHostApis out
pm_winmm_term called
pm_winmm_term exiting
Type ENTER...


C:\Users\Arjan\source\repos\performous\performous2\build\x64-Debug\Performous.exe (process 14592) exited with code -1.
To automatically close the console when debugging stops, enable Tools->Options->Debugging->Automatically close the console when debugging stops.
Press any key to close this window . . .
```

Do note the same is done in `audio.cc` if i'm correct.

**Comment by Lord-Kamina** [comment]
Even if it doesn't get rid of those messages... it seems the hack is no longer needed?
Have you investigated any more about that message?

File: game/libda/portaudio.hpp [hunk 1]
```diff
@@ -243,11 +243,7 @@ namespace portaudio {
 		}
 		~Stream() {
 			if (!m_handle) return;
-			// Give audio a little time to shutdown but then just quit
-			auto audiokiller = std::async(std::launch::async, Pa_CloseStream, m_handle);
-			if (audiokiller.wait_for(std::chrono::seconds(5)) == std::future_status::ready) return;
-			std::cerr << "PortAudio BUG: Pa_CloseStream hung for more than five seconds. Aborting." << std::endl;
-			abort();  // Crash. Calling exit() is prone to hang.
+			PORTAUDIO_CHECKED(Pa_CloseStream, (m_handle));
```
Comment (OznOg) [inline game/libda/portaudio.hpp:246]: what is this supposed to do? could you give pointers to documentation, I don't find it

**Comment by OznOg** [comment]
@Baklap4  would be nice to have the backtrace of all threads if possible

**Comment by OznOg** [comment]
https://github.com/clickteam-plugin/PortAudio/blob/master/PortAudio_SDK/src/hostapi/wasapi/pa_win_wasapi.c
not really clear but comment seems to check that "		if (g_WasapiInitThread != calling_thread_id)" ;
I would say it checks that the thread that did the init of PA is the same that finished PA. seems to be a windows specific thing

**Comment by Baklap4** [comment]
> @Baklap4 would be nice to have the backtrace of all threads if possible

At the time of breaking there's only the mainthread running

**Comment by OznOg** [comment]
> > @Baklap4 would be nice to have the backtrace of all threads if possible
> 
> At the time of breaking there's only the mainthread running

so the main thread changed id? something seems awkward in here...

**Comment by OznOg** [comment]
Did you pay attention that there are actually 2 audio killers, one on stream, one on audio itself. Seems that this patch removes the one on stream, but the one performing PA init in actually in game/audio.cc in the close() function.
@Baklap4 could you retest when removing the one in close() ?

**Comment by OznOg** [comment]
@Baklap4 is there a chance for you to retest with the later version?

**Comment by Baklap4** [comment]
I'll test to see if it runs tonight. Any specific things you want me to check?

**Review by Baklap4 — APPROVED** [review]
Not seeing the error anymore upon closing Performous.
LGTM

**Pull request merged**